### PR TITLE
explaintest: add explain test for partition pruning

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve? 

Add a explain test for partition pruning.

### What is changed and how it works?

TiDB explain result is different from mysql, so we have the explain test.

There are still many SQL statements that could take the advantage of partition pruning in MySQL,
but can't be pruned in TiDB. It's a known issue and I'll resolve it later.

Some statement doesn't work in TiDB as MySQL:

```
show status like 'Handler_read_rnd_next';
show status like 'Handler_read_key';
```

The result is generated using command:

```
./run-tests.sh -r partition_pruning
```

I manually adjust some test input adding `order by` as partitioned table doesn't keep result order.

### Check List 

Tests 

 - Integration test
and 
 - Manual test (add detailed scripts or steps below)

@zz-jason @coocood

### CODE REVIEW

**Comment by shenli** [comment]
Is there any issue for this:

> There are still many SQL statements that could take the advantage of partition pruning in MySQL,
but can't be pruned in TiDB. It's a known issue and I'll resolve it later.

File: plan/explain.go [hunk 1]
```diff
@@ -34,6 +34,12 @@ func (p *PhysicalIndexScan) ExplainInfo() string {
 		tblName = p.TableAsName.O
 	}
 	fmt.Fprintf(buffer, "table:%s", tblName)
+	if p.isPartition {
+		if pi := p.Table.GetPartitionInfo(); pi != nil {
```
Comment (shenli) [inline plan/explain.go]: Is there any chance that pi is nil?

File: model/model.go [hunk 2]
```diff
@@ -284,6 +284,15 @@ type PartitionInfo struct {
 	Definitions []PartitionDefinition `json:"definitions"`
 }
 
+func (pi *PartitionInfo) GetNameByID(id int64) string {
+	for _, def := range pi.Definitions {
+		if id == def.ID {
+			return def.Name.L
+		}
+	}
+	return ""
```
Comment (shenli) [inline model/model.go]: Should we return error here?

Comment (tiancaiamao) [reply model/model.go]: Not necessary.
The only place we need to know the partition name is `explain`, while `explain` doesn't care `GetNameByID` fail.

File: plan/explain.go (hunk 1 above)
Comment (tiancaiamao) [reply plan/explain.go]: `pi` should never be nil here...

File: model/model.go (hunk 2 above)
Comment (ciscoxll) [reply model/model.go]: Can it be changed to `notFind`?  empty strings that don't make sense are bad.

Comment (tiancaiamao) [reply model/model.go]: empty string means we have to check the code and fix bug, `notFind` gives no more information than that.

```
table: t, partition: 
```

It's a bug!

```
table: t, partition: notFind
```

It's a bug!

**Comment by zz-jason** [comment]
@tiancaiamao Please resolve conflicts.

**Comment by zz-jason** [comment]
LGTM

**Comment by zz-jason** [comment]
@tiancaiamao please fix ci

**Comment by tiancaiamao** [comment]
@zz-jason

**Pull request merged**